src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,69 @@
+import cn from 'classnames';
+import { useContext } from 'react';
+import { Todo } from '../../types/Todo';
+import { removeTodo } from '../../api/todos';
+import { TodosContext } from '../TodosContext/TodosContext';
+import { ErrorMessage } from '../../types/ErrorMessages';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    todos,
+    setTodos,
+    handleErrorMessage,
+  } = useContext(TodosContext);
+
+  const handleRemoveTodo = () => {
+    setTodos(todos.filter(value => value.id !== todo.id));
+
+    removeTodo(todo.id)
+      .then()
```

Comment:
update state only if request success
```suggestion

    removeTodo(todo.id)
      .then(() => setTodos(todos.filter(value => value.id !== todo.id)))
```